Description: Fix crash when using framebuffer renderer
Author: Ray Strode <rstrode@redhat.com>
Origin: upstream
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=80553
Applied-Upstream: http://cgit.freedesktop.org/plymouth/commit/?id=84eb4381db85877a9a56b35994e6c10d43e46ebe
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/libply-splash-core/ply-seat.c
+++ b/src/libply-splash-core/ply-seat.c
@@ -154,14 +154,28 @@
       add_pixel_displays (seat);
 
     }
-  else
+  else if (seat->terminal != NULL)
     {
       seat->keyboard = ply_keyboard_new_for_terminal (seat->terminal);
     }
-  add_text_displays (seat);
+  if (seat->terminal != NULL)
+    {
+      add_text_displays (seat);
+    }
+  else
+    {
+      ply_trace ("not adding text display for seat, since seat has no associated terminal.");
+    }
 
-  ply_keyboard_watch_for_input (seat->keyboard);
-  seat->keyboard_active = true;
+  if (seat->keyboard != NULL)
+    {
+      ply_keyboard_watch_for_input (seat->keyboard);
+      seat->keyboard_active = true;
+    }
+  else
+    {
+      ply_trace ("not watching seat for intpu");
+    }
 
   return true;
 }
